### ISSUE TITLE
Youtube download speed still slow

### ISSUE THREAD

**Issue by ahm750** [ytdl-org/youtube-dl#30398]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Description

It's great to see that youtube-dl has finally received an update. However, the slow download speed issue is still present. @dstftw Do you plan to update youtube-dl to fix this problem?

**Comment by Fritskee** [comment]
Issue has been addressed before, fixes have also been provided. I had the same issue and installing this merge request fixed it: `pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge`

Getting solid 5MiB/s now

**Comment by alakani** [comment]
The fact that a vaguely useful project ever existed here is merely coincidental; this project was created solely as a way for its malignant narcissist maintainer to use more people as a source of narcissistic supply. He has time to troll his own users every single day but apparently no time to do releases in 6 months. Brew install yt-dlp it's a fork by people who actually give a crap, all the bugs that dstftw uses as an excuse to attack and gaslight users were fixed in the other port many many months ago; speeds are good, the insane command line options make sense now and the defaults are actually useful, and thinking about using it doesn't make me want to decapitate myself. YMMV

**Comment by Fritskee** [comment]
@alakani fun guy!

**Comment by ahm750** [comment]
> Issue has been addressed before, fixes have also been provided. I had the same issue and installing this merge request fixed it: `pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge`
> 
> Getting solid 5MiB/s now

What about the Windows binary?

**Comment by Fritskee** [comment]
> > Issue has been addressed before, fixes have also been provided. I had the same issue and installing this merge request fixed it: `pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge`
> > Getting solid 5MiB/s now
> 
> What about the Windows binary?

Idk man, I don’t do windows. Just pip install it in a virtual environment…

**Comment by FranciscoPombal** [comment]
Just use https://github.com/yt-dlp/yt-dlp instead.

The current maintainer has proven himself to be incapable of properly steering this project.

Admittedly, this is a hard thing to do for a project of this size and reach, but by no means an insurmountable task.
However, the total refusal to seek help, delegate, and communicate, resulting in a period of 6 month-long radio silence followed a paltry release that doesn't fix the most pressing issue (throttling) without further explanations or apologies in such a crucial project is just inexcusable.

It is irresponsible and damaging to the FOSS world to run an important project like this.
Less technical users who don't keep up with every little bit of news haven't heard about yt-dlp, and so have been stuck with a barely-functioning youtube-dl release for months.

At this point the maintainer should cede the repository to the maintainers of yt-dlp, so that they can continue its development under the better-known `youtube-dl` "brand"/name.
The last thing that we need is for youtube-dl to become another Apache OpenOffice: [1](https://blog.documentfoundation.org/blog/2020/10/12/open-letter-to-apache-openoffice/), [2](https://twitter.com/hanno/status/1101433753824608257).

**Comment by 89z** [comment]
@FranciscoPombal shut up.

**Comment by fastily** [comment]
It's over folks.  dstftw has [removed](https://github.com/ytdl-org/youtube-dl/commit/21b759057502c6e70d51011cfb3fb86d84055182) himself from the list of active maintainers.  Might as well consider this repository done and archived at this point

**Comment by ugjka** [comment]
Open source is a cut throat business and no one is getting paid

**Comment by dirkf** [comment]
> Issue has been addressed before, fixes have also been provided. I had the same issue and installing this merge request fixed it: `pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge`

The PR was updated from 2021.12.17 and modified to use the more general unthrottling approach from yt-dlp. The single-file update is unchanged, ie includes both the original unthrottling fix and the fix from 2021.12.17.

**Comment by 89z** [comment]
I have an alternative tool that works with Windows:

https://github.com/89z/mech/releases/tag/v1.42.0

**Comment by RichCaloggero** [comment]
Having same issue. It's pretty clear IMO that google / youtube is throttling. If they wanted us to be able to download content, they would give us a download button, or other way via the UI to download. Just the facdt that this program and others like it exist should tell you that we're in an arms race with one of the most powerful corporations on the planet, and at the moment, we're losing.

Remember to always check every binary you install for malware!

**Comment by FranciscoPombal** [comment]
> It's over folks.  dstftw has [removed](https://github.com/ytdl-org/youtube-dl/commit/21b759057502c6e70d51011cfb3fb86d84055182) himself from the list of active maintainers.  Might as well consider this repository done and archived at this point

@dstftw 

Please consider collaborating with the yt-dlp maintainers and put them in charge of this repository, so that they can continue development of their active fork under the better-known name.

Alternatively, set this repository to arrived statusand place a big notice at the top of the README stating that it is abandoned, but further development continues on the yt-dlp fork.

Your efforts in the maintenence of the project are much appreciated, but you can't (well, shouldn't) abandon a project without ensuring all users know where they can go next - it's the least you can do when stepping down as a maintainer.

**Comment by RichCaloggero** [comment]
yt-dlt does seem to fix the issue, and doesn't seem to have installed malware.

https://github.com/yt-dlp/yt-dlp

**Comment by darksystem23** [comment]
> > > Issue has been addressed before, fixes have also been provided. I had the same issue and installing this merge request fixed it: `pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge`
> > > Getting solid 5MiB/s now
> > 
> > 
> > What about the Windows binary?
> 
> Idk man, I don’t do windows. Just pip install it in a virtual environment…

Thanks @Fritskee , your proposal seems to work. Unfortunately, Fedora decided to stop maintaining this library up-to-date. Last one on Fedora 35 from June... I've seen many people complaining about this issue, but sometimes it's just about how your distribution is maintaining the library. However, yt-dlp is up-to-date, so it seems the Fedora team is supporting the alternative.

Thanks :)

**Comment by andyncake** [comment]
> I have an alternative tool that works with Windows:
> 
> https://github.com/89z/mech/releases/tag/v1.42.0

I admire your bravery and shamelessness in promoting your **useless** project **everywhere**.

**Comment by 89z** [comment]
> I admire your bravery and shamelessness in promoting your useless project everywhere.

@andyncake fuck you. If YT-DLP can do it, so can I.

**Comment by dirkf** [comment]
Someone had a bad day?

As we all know, yt-dlp is a fork that shares code and a lot of UI with yt-dl. If yt-dlp can do something that yt-dl ought to do, it's normally possibly to back-port the code, and installing yt-dlp may also be a drop-in solution for some users.

It isn't generally helpful in a GH support forum to recommend an alternative program, since contributors are trying improve the subject program. There are other places to discuss various (in this case) media archiving tools.

A comment describing a technique that could be applied to improve the subject program is a quite reasonable opportunity to mention how the technique is successfully implemented in a similar program.

**Comment by 89z** [comment]
Watch out everyone, issue police is here!

# 👮🏻‍♂️🚓🚔🚨

**Comment by dirkf** [comment]
>Watch out everyone, issue police is here!

Nice emojis but let's not obscure that the post replaced a humourless insult responding to @andyncake, one that reflected so poorly on the author that I didn't quote it. Anyone who has run or contributed to software projects can distinguish useful contributions from marginal trolling.

Or perhaps @89z is a New York cop outside GH?

Lady tourist to cop: "Officer, can you tell me the way to Times Square, please, or should I just go fuck myself?"

[That's how I remember the gag from quite a long time before [all of these versions](https://www.barrypopik.com/index.php/new_york_city/entry/do_you_know_the_way_to_the_empire_state_building_or_should_i_just_go_fuck_m).]

**Comment by 89z** [comment]
> let's not obscure that the post replaced a humourless insult responding to @andyncake, one that reflected so poorly on the author that I didn't quote it.

@dirkf while we are rewriting history, maybe we can also omit the part where you made a pointless off topic joke [1]. Or are you the only one allowed to make off topic rude comments?

> Anyone who has run or contributed to software projects can distinguish useful contributions from marginal trolling.

Remind me again where your contributions are, I am not seeing anything [2], or are you referring to your open PRs on a dead repo [3]?

1. https://github.com/ytdl-org/youtube-dl/issues/30442#issuecomment-1002811574
2. https://github.com/dirkf?tab=repositories&type=source
3. https://github.com/ytdl-org/youtube-dl/pulls?q=author:dirkf

**Comment by fstirlitz** [comment]
> one that reflected so poorly on the author that I didn't quote it.

Not to worry, nothing gets lost on the Internet…

<details> <summary> Screenshot from OctoDroid.</summary>

![89z](https://user-images.githubusercontent.com/4331184/147829880-bd093735-2d13-4e1e-ba61-467a3c97c086.png)

I would assume this is generated from GitHub’s public API. Anyone can open [the commenter’s page](https://github.com/89z) in the same app and verify.

</details>

Perhaps [Steven Penny from Dallas](https://archive.md/9vP7p) should think twice before he writes. I would not hope to get hired with _that_ attitude.

**Comment by 89z** [comment]
@fstirlitz fuck you too!

**Comment by 89z** [comment]
I have just reported @fstirlitz for doxing. I dont know what youre trying to accomplish by posting my personal information, but I suggest you stop it.

**Comment by dirkf** [comment]
> .... doxxing ...

What, [like Google](https://www.google.com/search?q=srpen6@gmail.com)?

**Comment by 89z** [comment]
I have now reported @dirkf as well for doxing. Really I would suggest you stop it, as your comments have nothing to do with the topic on this page.

**Comment by kssi** [comment]
Steve, as per [Wikipedia](https://en.wikipedia.org/wiki/Doxing):

> Doxing or doxxing is the act of publicly revealing **previously private personal information** about an individual or organization

> Essentially, doxing is revealing and publicizing the records of an individual, **which were previously private or difficult to obtain**.

I suggest you stop being so painfully wrong about everything you do and state.

**Comment by kssi** [comment]
Steven, as per [Wikipedia](https://en.wikipedia.org/wiki/Doxing):

> Doxing or doxxing is the act of publicly revealing **previously private personal information** about an individual or organization

> Essentially, doxing is revealing and publicizing the records of an individual, **which were previously private or difficult to obtain**.

I suggest you stop being so painfully wrong about everything you do and state.

**Comment by 89z** [comment]
@kssi ah yes, Wikipedia, the great source of information:

https://www.youtube.com/watch?v=kFBDn5PiL00

**Comment by alakani** [comment]
> personal information

Your email address is on your public profile, the github logs are also public and you would have to take it up with them. You're profile also implies that you own zombo.com - I don't think Josh would appreciate that, especially with how you're acting

**Comment by 89z** [comment]
> Your email address is on your public profile, the github logs are also public and you would have to take it up with them. Your profile also implies that you own

@alakani the name and location @fstirlitz listed is not on my profile, nor any comment about me getting hired. So he purposefully searched that from some other location, in order to shame me into stop commenting. I would say thats pretty clear doxing behavior, and shouldnt be tolerated here or anywhere. I only changed my info to Zombo because of creeps like @fstirlitz stalking me online. They dont have any valid on topic counters, so they resort to doxing and online shaming.

**Comment by kssi** [comment]
> the name and location fstirlitz listed is not on my profile, nor any comment about me getting hired.

It was just yesterday. And liar on top of that.

**Comment by alakani** [comment]
> name and location fstirlitz listed is not on my profile

They're being petty because they didn't like you spamming and attacking people for no reason, but it's still public information. You posted it all yourself to public websites. If you're ashamed of stuff that you posted yourself, well....

> changed my info to Zombo because

A website or twitter are not mandatory here, no need to put somebody else's info on your profile

**Comment by 89z** [comment]
This all started because of @dirkf shitty comment [1], which linked to someone elses shitty comment [2]. I may not have responded the best way, but the point I wanted to make was that my project is not useless. My project has a fix for the issue on this page, so that seems useful to me.

I shouldnt really even care what @andyncake has to say, as from his profile he just seems to be a shitty GitHub troll with no actual contributions. At any rate, Ive made my point. If anyone else want to contest the usefulness of my project fine, or about the YouTube speed in the original post. But any comments broadcasting my personal information arent really on topic.

1. https://github.com/ytdl-org/youtube-dl/issues/30442#issuecomment-1002811574
2. https://github.com/ytdl-org/youtube-dl/issues/30398#issuecomment-1001226243

**Comment by alakani** [comment]
> my project is not useless

Yes but it has very little documentation to tell people if it will be useful for their particular situation. You've posted usage examples in random comments, but not on the project page. It doesn't work if you just yell at people and tell them it's good, you'd have to show them. It's better if users tell people about the project, if the maintainer does it many times per day, it's spam

> no actual contributions

I don't have any public either, unfortunately. Some day.

> arent really on topic

And telling everybody to copulate with themselves and consume genitalia is?

**Comment by 89z** [comment]
> Yes but it has very little documentation to tell people if it will be useful for their particular situation. You've posted usage examples in random comments, but not on the project page. It doesn't work if you just yell at people and tell them it's good, you'd have to show them. It's better if users tell people about the project, if the maintainer does it many times per day, it's spam

I dont think I need advice from someone who doesnt have any repos.

**Comment by alakani** [comment]
> dont think I need advice

We're trying to help you not get banned from github entirely, you've violated the TOS several times

**Comment by 89z** [comment]
> We're trying to help you not get banned from github entirely, you've violated the TOS several times

@alakani thanks for the help. You have violated as well, so you should be careful.

**Comment by megapro17** [comment]
Ahahahah he deleted all his replies. can anyone share what was there? it should be in email, if you replied before

**Comment by fstirlitz** [comment]
- <https://archive.ph/https://github.com/ytdl-org/youtube-dl/issues/30398>
- <https://archive.ph/H7h26>

(I guess we owe an apology to the original author of this ticket, who probably didn’t open it just to be witness to this silly spat.)

**Comment by megapro17** [comment]
@fstirlitz Oh, sorry, I checked archive org, but forgot about this. Thanks

**Comment by dirkf** [comment]
>(I guess we owe an apology to the original author of this ticket, who probably didn’t open it just to be witness to this silly spat.)

@alakani,  @fstirlitz, @kssi, @megapro17
As many comments that were being responded to have _magically disappeared_, shouldn't we just hide our spat-related posts as off-topic?

**Comment by megapro17** [comment]
What's the point, if issue will not be resolved anyway here, and it's fixed long time ago in fork.

**Comment by jk-85** [comment]
>  `pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge`
I tried that command but it sais "Did not find branch or tag 'refs/pull/30184/merge', assuming revision or ref." and then installs an old version (2021.06.06.) What can I do?

**Comment by dirkf** [comment]
Your _pip_ may be [too old](https://stackoverflow.com/a/50095199).

1. Try [_pipx_](https://github.com/ytdl-org/youtube-dl/pull/30184#issuecomment-997235883), or just get the latest commit from the PR (at the time of writing) instead:
```
pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@905d1d281ddfa5d183fc445010d350cefc6a58ec
```

**Comment by 20NE** [comment]
come on, just merge it, gdi

**Comment by dirkf** [comment]
[Quite](https://github.com/ytdl-org/youtube-dl/commit/5014bd67c22b421207b2650d4dc874b95b36dda1#commitcomment-62060781), [but ...](https://github.com/ytdl-org/youtube-dl/commit/21b759057502c6e70d51011cfb3fb86d84055182).

**Comment by jk-85** [comment]
Oops, yes, you're right, thank you, my _pip_ was too old.

**Comment by AndNovAtor** [comment]
Hey guys:

> Closed with merge of [57044ea](https://github.com/ytdl-org/youtube-dl/commit/57044eacebc6f2f3cd83c345e1b6e659a22e4773)..[af9e725](https://github.com/ytdl-org/youtube-dl/commit/af9e72507ea38e5ab3fa2751ed09ec88021260cb).

Yupi!

**Comment by nilsonsales** [comment]
> Hey guys:
> 
> [#30184 (comment)](https://github.com/ytdl-org/youtube-dl/pull/30184#issuecomment-1025261055)
> 
> > dirkf wroted:
> > Closed with merge of [57044ea](https://github.com/ytdl-org/youtube-dl/commit/57044eacebc6f2f3cd83c345e1b6e659a22e4773)..[af9e725](https://github.com/ytdl-org/youtube-dl/commit/af9e72507ea38e5ab3fa2751ed09ec88021260cb).
> 
> Yupi!

Hey! Do you know how can I download this latest version? I've tried both via pip or curl latest (as in the instructions), but I can only get the version from 2021.12.17.

**Comment by garoto** [comment]
> only get the version from 2021.12.17.

version date is only updated when generating a new release:

https://github.com/ytdl-org/youtube-dl/blob/master/devscripts/release.sh#L74-L75

**Comment by HIBFITAS** [comment]
I don't know if it has been fixed already, but aria2c seems to ease the pain for me. 
I use aria2c to start like 10 connections to the server and it will merge them down to a file.
I didn't try with more than 10. I don't know what the max is, but anyway it should be 10 x 70kb.

This is what works for me in Windows 10 – cmd-box:

youtube-dl.exe -url "https://youtu.be/dvV6x2BQoXk" --username=yourusername@gmail.com --password=yourpassword --audio-quality 256K --external-downloader aria2c --external-downloader-args "-c -j10 -x10 -s10 -k1M"

These are youtube-dl switches:
-url  --username  --password  --audio-quality 256K  --external-downloader  --external-downloader-args

These are aria2c switches:
"-c -j10 -x10 -s10 -k1M"

Please refer to both youtube-dl and aria2c docs for their command-line switches.
--hope it help those who didn't upgrade (like me :) )--

**Comment by mikewesten** [comment]
> Issue has been addressed before, fixes have also been provided. I had the same issue and installing this merge request fixed it: `pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge`
> 
> Getting solid 5MiB/s now
gives me this
`$ pip install --force-reinstall -U git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge
Collecting git+https://github.com/ytdl-org/youtube-dl@refs/pull/30184/merge
  Cloning https://github.com/ytdl-org/youtube-dl (to revision refs/pull/30184/merge) to /tmp/pip-req-build-qi7fdes3
  Running command git clone -q https://github.com/ytdl-org/youtube-dl /tmp/pip-req-build-qi7fdes3
  WARNING: Did not find branch or tag 'refs/pull/30184/merge', assuming revision or ref.
  Running command git fetch -q https://github.com/ytdl-org/youtube-dl refs/pull/30184/merge
  fatal: couldn't find remote ref refs/pull/30184/merge
ERROR: Command errored out with exit status 128: git fetch -q https://github.com/ytdl-org/youtube-dl refs/pull/30184/merge Check the logs for full command output.`

**Comment by dirkf** [comment]
You can install from the HEAD instead: the PR has been merged. I hope that means just leaving off the @... but check the _pip_ documentation.

**Comment by AndNovAtor** [comment]
@dirkf , I want to note some interesting behavior. Current `master` is works fine, all good. But if I use NOW "drop-in-replacement" file from https://github.com/ytdl-org/youtube-dl/pull/30184  ([your direct link](https://github.com/ytdl-org/youtube-dl/raw/3b9b1cd4314f9d44630a2a65abfff8abab454887/youtube_dl/extractor/youtube.py) ), it's slow "again". And before ~**12.02.2022** - all was fine too with "drop-in-replacement" file.

So, I used "drop-in-replacement" file, to "modify" my youtube-dl installed by pip (I didn't use installing by pip from "merge"), simply download this file, backup "old" file in Python lib dir, paste "new" file there). SImply. Since 2021.06.06 version (I used it since ~12.2021). All worked fine. Till ~**12.02.2022**.

Now, if I use stable release (from pip) with replaced `youtube.py` file, I got this multiple (~23 lines) "error":
`WARNING: [youtube] Couldn't extract YouTube video throttling parameter descrambling function name`
Than downloading "fallback" to simple downloading, so speed again is ~50kb/s

Again: I tried lastest dev version on master (installed  to "dedicated" "clean" virtualenv by `pip install git+https://github.com/ytdl-org/youtube-dl`) - and all works fine, normal fast speed.

I simply note this behavior. No actions is needed, old "issue" "fixed" on master branch.
